blockstack-sdk/src/main/java/org/blockstack/android/sdk/Blockstack.kt

```diff
@@ -519,17 +533,21 @@ class Blockstack(private val callFactory: Call.Factory = OkHttpClient(),
         }
 
         val issuerPublicKey = payload.getJSONObject("issuer").getString("publicKey")
-        val uncompressedAddress = issuerPublicKey.toBtcAddress()
+        val uncompressedBtcAddress = issuerPublicKey.toBtcAddress()
+        val uncompressedStxAddress = issuerPublicKey.toStxAddress(true)
 
-        if (publicKeyOrAddress == issuerPublicKey) {
-            // pass
-        } else {
-            if (publicKeyOrAddress == uncompressedAddress) {
+            if (publicKeyOrAddress == issuerPublicKey) {
                 // pass
             } else {
-                throw Error("Token issuer public key does not match the verifying value")
+                when (publicKeyOrAddress) {
+                    uncompressedBtcAddress -> { /* Pass */}
+                    uncompressedStxAddress -> { /* Pass */}
+                    else -> {
+                        throw Error("Token issuer public key does not match the verifying value")
+                    }
+                }
```

Comment:
Feels weird using a `when` to check if one variable is different from 2 other variables. Feels like an `if` would be clearer, but it might just be because it's the first time I've seen this.